### PR TITLE
Restore modules from the NuGet package cache by using dotnet restore

### PR DESCRIPTION
**Pull request by daxian-dbw**
## PR Summary

Restore modules from the NuGet package cache by using `dotnet restore`.
This is to bring back the build related changes in PR #5759  /cc @lzybkr 

## PR Checklist

Note: Please mark anything not applicable to this PR `NA`.

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [x] Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] User facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - [ ] Issue filed - Issue link:
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
    - [x] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.

### CODE REVIEW

**Comment by adityapatwardhan** [comment]
Should the PSReadline files be removed? We are still not getting PSReadline from gallery?

**Comment by daxian-dbw** [comment]
We are not getting PSReadLine from gallery yet. This is just the build part changes from the original PR.

File: test/PSReadLine/App.config [hunk 1]
```diff
@@ -1,6 +0,0 @@
-﻿<?xml version="1.0" encoding="utf-8" ?>
-<configuration>
-    <startup> 
-        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.5" />
```
Comment (TravisEz13) [inline test/PSReadLine/App.config]: why are these files being remove?

Comment (lzybkr) [reply test/PSReadLine/App.config]: It's reasonable to remove these files at any time - this code is just a simple console application for quick manual testing of PSReadLine.

**Review by TravisEz13 — APPROVED** [review]

**Comment by daxian-dbw** [comment]
Resolved conflicts and change to restore Pester also using NuGet package cache.

**Review by TravisEz13 — APPROVED** [review]

**Comment by TravisEz13** [comment]
https://travis-ci.org/PowerShell/PowerShell/jobs/346459732#L7109

please fix test failures

**Comment by TravisEz13** [comment]
I'm working on the test failures

**Comment by daxian-dbw** [comment]
The failures in Travis CI were known issue and have been fixed via #6249
I have restarted the Travis CI.

**Pull request merged**